src/js-form-validator.test.ts

```diff
@@ -85,3 +85,31 @@ QUnit.test('should fail email validation with special characters in domain', ass
 
   assert.equal(FormValidator.Validate(inputElement), false);
 });
+
+QUnit.test('should fail checkbox validation when checked state is not set', assert => {
+  inputElement.setAttribute('type', 'checkbox');
+  inputElement.setAttribute('data-validate', 'checkbox');
+
+  assert.equal(FormValidator.Validate(inputElement), false);
+});
+
+QUnit.test('should fail checkbox validation when checked state set to false', assert => {
+  inputElement.setAttribute('type', 'checkbox');
+  inputElement.setAttribute('data-validate', 'checkbox');
+  inputElement.checked = false;
+
+  assert.equal(FormValidator.Validate(inputElement), false);
```

Comment:
If you'd like to use it, `QUnit` has special methods for checking booleans called `ok` and `notOk`. If you feel that is less readable though, I'm fine with how you have it.